third-party/rsutils/include/rsutils/json.h

```diff
@@ -124,6 +124,18 @@ inline std::ostream & operator<<( std::ostream & os, json_ref const & j )
 }
 
 
+// When you do:
+//     json j = j2.nested( "subkey" );
+// The nesting returns a json_ref, correctly.
+// operator=() get called. But the implementation is a copy-and-swap one, meaning it attempts to construct a json value
+// from the argument. Since the argument isn't a json object (doesn't matter that it has a conversion operator), it does
+// this by attempting to deserialize it with to_json()! We need to avoid this, for this usage and others...
+inline void to_json( rsutils::json & j, json_ref const & jr )
```

Comment:
This gets called automatically by the json constructor?

Reply:
Yes, they have a constructor to take any unknown type and convert it to json. It does this via `to_json`.